boot/seal.go

```diff
@@ -102,10 +103,16 @@ func recoveryBootChainsFileUnder(rootdir string) string {
 	return filepath.Join(dirs.SnapFDEDirUnder(rootdir), "recovery-boot-chains")
 }
 
+type sealKeyToModeenvFlags struct {
+	// FactoryReset indicates that the sealing is happening during factory
+	// reset.
+	FactoryReset bool
+}
+
 // sealKeyToModeenv seals the supplied keys to the parameters specified
 // in modeenv.
 // It assumes to be invoked in install mode.
-func sealKeyToModeenv(key, saveKey keys.EncryptionKey, model *asserts.Model, modeenv *Modeenv) error {
+func sealKeyToModeenv(key, saveKey keys.EncryptionKey, model *asserts.Model, modeenv *Modeenv, flags sealKeyToModeenvFlags) error {
```

Comment:
since you are passing this to sub-functions, maybe it's better to use a pointer (in case the struct grows bigger)?